src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -296,34 +297,33 @@ private function fixPhpDoc(Tokens $tokens, int $index, array &$uses, string $nam
 
         $phpDoc = $tokens[$index];
         $phpDocContent = $phpDoc->getContent();
-        Preg::matchAll('#@([^\s]+)(\s+)([a-zA-Z0-9_\\\\]+)#', $phpDocContent, $matches);
+        $phpDocContentNew = Preg::replaceCallback('/([*{]\h*@)(\S+)(\h+)('.TypeExpression::REGEX_TYPES.')(?!(?!\})\S)/', function ($matches) use ($allowedTags, &$uses, $namespaceName) {
+            if (!\in_array($matches[2], $allowedTags, true)) {
+                return $matches[0];
+            }
 
-        if ([] !== $matches[0]) {
-            foreach ($matches[3] as $i => $typeName) {
-                if (!\in_array($matches[1][$i], $allowedTags, true)) {
-                    continue;
-                }
+            // @TODO parse the complex type using TypeExpression and fix all names inside (like `int|string` or `list<int|string>`)
+            if (!Preg::match('/^[a-zA-Z0-9_\\\\]+(\|null)?$/', $matches[4])) {
```

Comment:
What about `null|...`? Why is the nullable part added here?

Reply:
We do not support unioned types yet, however, the first type in union is fixed correctly already, thus we allow `A|null` but not `null|A` as the later will fix only the `null`.

This is helpful to fix `@var A|null` which is quite common.

General support of union, intersect... is a feature request - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7619